cluster_library.c

```diff
@@ -1486,6 +1496,24 @@ PHP_REDIS_API void cluster_bulk_raw_resp(INTERNAL_FUNCTION_PARAMETERS,
     efree(resp);
 }
 
+PHP_REDIS_API void
+cluster_single_line_resp(INTERNAL_FUNCTION_PARAMETERS, redisCluster *c, void *ctx)
+{
+    char *p;
+
+    /* Cluster already has the reply so abort if this isn't a LINE response *or* if for
+     * some freaky reason we don't detect a null terminator */
+    if (c->reply_type != TYPE_LINE || !(p = memchr(c->line_reply,'\0',sizeof(c->line_reply)))) {
```

Comment:
I'm not sure about "freaky reason" check. Is it really necessary? Also `sizeof(c->line_reply)` is the size of pointer but not the length of string.

Reply:
I'm attempting to protect against a buffer overrun here, which I agree should be highly unlikely but not theoretically impossible.

The `memchr` check makes sure that we have a null terminator inside of our 1024 byte char array, which we should _always_ have if something has not gone horribly wrong.  😀

I have to make the check for the `\0` anyway to find the length of the string, and am not sure what a safe way to do it it would be if not this.